include/coap3/coap_debug.h

```diff
@@ -48,25 +48,27 @@ typedef short coap_log_t;
     with LOG_CIPHERS added. */
 
 #if !defined(RIOT_VERSION)
-typedef enum {
-  LOG_EMERG=0, /**< Emergency */
-  LOG_ALERT,   /**< Alert */
-  LOG_CRIT,    /**< Critical */
-  LOG_ERR,     /**< Error */
-  LOG_WARNING, /**< Warning */
-  LOG_NOTICE,  /**< Notice */
-  LOG_INFO,    /**< Information */
-  LOG_DEBUG,   /**< Debug */
-  COAP_LOG_CIPHERS=LOG_DEBUG+2 /**< CipherInfo */
-} coap_log_t;
+typedef int coap_log_t;
+#ifndef LOG_EMERG
```

Comment:
It might be cleaner if we moved these `#define`s outside the `#ifdef`s and guard every definition with its corresponding `#ifndef`. This way, RIOT and non-RIOT could be treated equally:

```c
#ifndef LOG_EMERG
#define LOG_EMERG   (0)
#endif
#ifndef LOG_ALERT
#define LOG_ALERT   (1)
#endif
```
...

Reply:
Updated as suggested and code pushed